packages/@aws-cdk/aws-codepipeline-actions/lib/stepfunctions/invoke-action.ts

```diff
@@ -0,0 +1,149 @@
+import * as codepipeline from '@aws-cdk/aws-codepipeline';
+import * as iam from '@aws-cdk/aws-iam';
+import * as stepfunction from '@aws-cdk/aws-stepfunctions';
+import { Construct} from '@aws-cdk/core';
+import { Action } from '../action';
+ 
+/**
+ * Represents the input type for the StateMachine.
+ */
+export enum StateMachineInputType {
+  /**
+   * When specified, the value in the Input field is passed 
+   * directly to the state machine input.
+   */
+  LITERAL = 'Literal',
+
+  /**
+   * The contents of a file in the input artifact specified by the Input field 
+   * is used as the input for the state machine execution. 
+   * Input artifact and FilePath is required when InputType is set to FilePath.
+   */
+  FILEPATH = 'FilePath',
+}
+
+/**
+ * Construction properties of the {@link StepFunctionsInvokeAction StepFunction Invoke Action}.
+ */
+export interface StepFunctionsInvokeActionProps extends codepipeline.CommonAwsActionProps {
+  /**
+   * The optional input Artifact of the Action.
+   * If InputType is set to FilePath, this artifact is required 
+   * and is used to source the input for the state machine execution.
+   * 
+   * @default the Action will not have any inputs
+   * @see https://docs.aws.amazon.com/codepipeline/latest/userguide/action-reference-StepFunctions.html#action-reference-StepFunctions-example
+   */
+  readonly input?: codepipeline.Artifact;
+ 
+  /**
+   * The optional output Artifact of the Action.
+   *
+   * @default the Action will not have any outputs
+   */
+  readonly output?: codepipeline.Artifact;
+ 
+  /**
+   * The state machine to invoke.
+   */
+  readonly stateMachine: stepfunction.IStateMachine;
+ 
+  /**
+   * Optional StateMachine InputType
+   * InputType can be Literal or FilePath
+   *
+   * @default - Literal
+   */
+  readonly stateMachineInputType?: StateMachineInputType;
+ 
+  /**y
+   * When InputType is set to Literal (default), this field is optional.
+   * If provided, the Input field is used directly as the input for the state machine execution.
+   * Otherwise, the state machine is invoked with an empty JSON object {}.
+   * 
+   * When InputType is set to FilePath, this field is required.
+   * An input artifact is also required when InputType is set to FilePath.
+   * 
+   * @default - none
+   */
+  readonly stateMachineInput?: string | object;
+ 
+ 
+  /**
+   * Prefix (optinoal)
+   *
+   * By default, the action execution ID is used as the state machine execution name. 
+   * If a prefix is provided, it is prepended to the action execution ID with a hyphen and 
+   * together used as the state machine execution name.
+   */
+  readonly executionPrefixName?: string;
+}
+ 
+/**
+ * CodePipeline invoke Action that is provided by an AWS Step Function.
```

Comment:
this comment didn't really make sense to me, suggest:

```suggestion
 * Step Function invoke Action that is provided by an AWS CodePipeline.
```